### ISSUE TITLE
Version 3 - leftovers

### ISSUE THREAD

**Issue by rpoleski** [rpoleski/MulensModel#155]
- [x] why `t_eff` is not in `ModelParameters._primary_source_params_head`?
- [x] What is the purpose of `Model._N_source_attr`?
- [x] review `documents/MM_v3.md`
- [x] move `documents/MM_v3.md` to v4
- [x] `magnificationcurve.py` - remove comment in `get_magnification()`
- [x] uncomment `tests/test_FitData.py::TestFSPLGradient2::test_dAdu_FSPLError`
- [x] uncomment or remove: `test_PointLensWithShear.py` class `test_errors`
- [x] parameters of methods should be checked and give warnings when they're don't match allow names

**Comment by rapoliveira** [comment]
> * [X] `magnificationcurve.py` - remove comment in `get_magnification()`

Commits 4f5bd52 and 38b39b3 apply this change as agreed in person. A new universal function calculates the magnification for both point lens and binary lens, to avoid code duplication.

> * [X] uncomment tests/test_FitData.py::TestFSPLGradient2::test_dAdu_FSPLError

Commits 95b09b8 and f7eb66c uncomment the test and add two similar tests, checking if NotImplementedError is raised when calling `get_d_A_d_u_for_[...]()` for different models.

> * [X] uncomment or remove: test_PointLensWithShear.py class test_errors

Done, the six functions tested for NotImplementedError do not even exist in `pointlenswithshear.py`.

@rpoleski Please let me know if something else should be done in these points. I will be working on the next ones.

**Comment by rpoleski** [comment]
> I'll look at other two later on.

I had a look - everything is OK.

**Comment by rapoliveira** [comment]
> What is the purpose of `Model._N_source_attr`?

It was added in the commit a7f64dc, by Jennifer.

As commented there, this specific variable is not used in the code and no tests fail if it is removed.
It just indicates that magnification curves are stored as `self._magnification_curve_1` and `self._magnification_curve_2`, as done in the function `_separate_magnifications()` of model.py.

@jenniferyee, should we leave or remove it?

**Comment by rapoliveira** [comment]
@rpoleski Yes, the changes are in this branch: [version3-leftovers](https://github.com/rapoliveira/MulensModel/tree/version3-leftovers).

The last three commits remove `_N_source_attr` and implement the last item about checking the parameters of magnification methods. Can I start a PR?

**Comment by rpoleski** [comment]
Thanks @rapoliveira for working on these issues.
